flow/project.py

```diff
@@ -1772,6 +1798,12 @@ def run(self, jobs=None, names=None, pretend=False, np=None, timeout=None, num=N
 
             The default value is `none`, which is equivalent to `by-job` in the current
             implementation.
+        :param ignore_conditions:
+            Specify if pre or post conditions check is to be ignored for eligibility check
+                * 'none' (default: all conditions are checked)
+                * 'pre' (ignore pre condition check)
+                * 'post' (ignore post condition check)
+                * 'all' (ignore all condition check)
```

Comment:
I'm not sure that a dual approach like this makes sense. I understand that we need to accept strings on the command line, but on the Python code level we could actually use the enum, right?

Reply:
I had the way you said implemented in a earlier commit where string is only used for terminal and Enum class is required for python `run` and `submit`. My main thoughts for moving to all string for user facing argument is this looks simpler and more unified for `run` and `submit` across using in python or terminal, and I thought it might be a bit too much requirement for user to be aware and use enum. But I want to get more thoughts on this and I am happy to change to whichever people prefers. @vyasr @b-butler

Reply:
I agree with Simon, everything in Python should use the enum. As long as the function docstring indicates that the enum should be used (with the appropriate Sphinx link to `~.IgnoreConditions`) it's fine to enforce that requirement.  Setting up an argparse Action like I showed you for the CLI is fine, but everything passed around in the code should be the enum object.